src/util/selinux.c

```diff
@@ -43,25 +44,47 @@
  */
 int selinux_file_context(const char *dst_name)
 {
+    struct selabel_handle *handle = NULL;
     char *scontext = NULL;
+    int ret;
 
     if (is_selinux_enabled() == 1) {
         /* Get the default security context for this file */
-        if (matchpathcon(dst_name, 0, &scontext) < 0) {
-            if (security_getenforce () != 0) {
-                return 1;
-            }
+        handle = selabel_open(SELABEL_CTX_FILE, NULL, 0);
+        if (handle == NULL) {
+            ret = errno;
+            DEBUG(SSSDBG_CRIT_FAILURE, "Unable to create selabel context "
+                  "[%d]: %s\n", ret, sss_strerror(ret));
+            ret = 1;
+            goto done;
+        }
+
```

Comment:
Hi,

`matchpathcon()` is calling `realpath()` on `dst_name` before calliing `selabel_lookup()`, this is mentioned in the man page as well. So it sounds important and we might want to call it as well.

bye,
Sumit

Reply:
> Hi,
> 
> `matchpathcon()` is calling `realpath()` on `dst_name` before calliing `selabel_lookup()`, this is mentioned in the man page as well. So it sounds important and we might want to call it as well.
> 
> bye,
> Sumit

Maybe I am wrong but `matchpathcon()` looks to be deprecated according to MAN:

```
This family of functions is deprecated.  For new code, please use
selabel_open(3) with the SELABEL_CTX_FILE backend in place of
matchpathcon_init(), use selabel_close(3) in place of
matchpathcon_fini(), and use selabel_lookup(3) in place of
matchpathcon().
```

Reply:
> > Hi,
> > `matchpathcon()` is calling `realpath()` on `dst_name` before calliing `selabel_lookup()`, this is mentioned in the man page as well. So it sounds important and we might want to call it as well.
> > bye,
> > Sumit
> 
> `matchpathcon()` is marked as deprecated so I think we should not use it as an example.
> 
> ```
> This family of functions is deprecated.  For new code, please use
> selabel_open(3) with the SELABEL_CTX_FILE backend in place of
> matchpathcon_init(), use selabel_close(3) in place of
> matchpathcon_fini(), and use selabel_lookup(3) in place of
> matchpathcon().
> ```

ah, sorry for not being specific here, I didn't meant to call `matchpathcon()` (I know this patch is about replacing `matchpathcon()` because it is deprecated) but to add a call to `realpath()`.

bye,
Sumit

Reply:
I'm waiting for an answer from selinux maintainer.

Reply:
@sumit-bose Good catch. "realpath" is needed because selabel_lookup doesn't recognise relative paths.

Reply:
Thank you. Done.